### PR TITLE
Reorganized server

### PR DESCRIPTION
**Pull request by Ozymadias**

### CODE REVIEW

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/Server.java [hunk 1]
```diff
@@ -3,9 +3,11 @@
 import com.epam.solid.nie.client.ui.Cell;
 
 public interface Server {
-    void connect(String ip);
+    boolean connect(String ip);
     boolean canConnect(String ip);
     void passAllShips(String allShips);
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/communication/Server.java]: napewno ma zwracac void?

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java [hunk 2]
```diff
@@ -3,9 +3,15 @@
...
 
-import java.util.Random;
+import java.io.IOException;
+import java.util.*;
 
 public class SocketServer implements Server {
+
+    private ShipClient server;
+    private String allMoves = "";
+    private Queue<Cell> cells = new LinkedList<>();//Arrays.asList(new Cell(Point2D.of(0, 0)), new Cell(Point2D.of(9, 9))));
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java]: remove comment

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java [hunk 3]
```diff
@@ -37,15 +37,16 @@
                 typesOfShips.poll();
                 if (typesOfShips.isEmpty()) {
                     socketServer.passAllShips(playerBoard.getAllpositions());
-                    //todo: method to place ships is called here!!!!!!
-                    running = placeShipsRandomly();
+                    //running = placeShipsRandomly();
+                     running = placeShipsRandomly(socketServer.receiveAllShips());
+//                    running = placeShips();
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java]: remove comments

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java [hunk 4]
```diff
@@ -61,24 +62,41 @@ private void shipOrientation(MouseEvent event) {
...
+            String[] coords = shipStr.split(",");
+            for (int i = 0; i < coords.length - 1; i += 2) {
+                int x = Integer.valueOf(coords[i]);
+                int y = Integer.valueOf(coords[i + 1]);
+                point2DOfShip.add(Point2D.of(x, y));
             }
+            shipCreator = createShip(coords);
+            Ship ship = new Ship(shipCreator.createBattleShip(point2DOfShip));
+            enemyBoard.isShipPositionValid(ship, new Cell(point2DOfShip.get(0)));
+            //System.out.println(Arrays.toString(coords));
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java]: comments

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java [hunk 5]
```diff
@@ -61,24 +62,41 @@ private void shipOrientation(MouseEvent event) {
...
         }
+        //return enemyBoard.allShips;
         return true;
     }
-    //todo: dominik
+
+    String createShip(String[] coords, boolean v) {
+        if (coords.length > 2 && Integer.valueOf(coords[0]).equals(Integer.valueOf(coords[2])))
+            return "vertical";
+//            return new ShipCreator(new VerticalShipFactory());
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java]: even more

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java [hunk 6]
```diff
@@ -61,24 +62,41 @@ private void shipOrientation(MouseEvent event) {
...
+            Ship ship = new Ship(shipCreator.createBattleShip(point2DOfShip));
+            enemyBoard.isShipPositionValid(ship, new Cell(point2DOfShip.get(0)));
+            //System.out.println(Arrays.toString(coords));
         }
+        //return enemyBoard.allShips;
         return true;
     }
-    //todo: dominik
+
+    String createShip(String[] coords, boolean v) {
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java]: another case is that we don't want method just for tests in production code

File: battleship-server/src/main/java/com.epam.solid.nie/server/server/Player.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package com.epam.solid.nie.server.server;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+public interface Player {
+    void register(ServerSocket serverSocket) throws IOException;
+
+    void inform(String s) throws IOException;
```
Comment (rivalez) [inline battleship-server/src/main/java/com.epam.solid.nie/server/server/Player.java]: hmm IMO inform isn't the best choice for method in player interface. Maybe try to rename it.

File: battleship-server/src/main/java/com.epam.solid.nie/server/server/NetPlayer.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Override
+    public String provideShips() throws IOException {
+        out.println("Provide ships");
+        return in.readLine();
+    }
+
+    @Override
+    public String makeMove() throws IOException {
+        //out.println("Make move");
```
Comment (rivalez) [inline battleship-server/src/main/java/com.epam.solid.nie/server/server/NetPlayer.java]: comment to remove

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    private PrintWriter out;
+    private BufferedReader in;
+    private String enemyShips;
+
+    SocketClient(String ip) {
+        this.ip = ip;
+    }
+
+    public boolean run() throws IOException {
+        int portNumber = 8080;
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: move to class level

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java [hunk 10]
```diff
@@ -15,26 +24,78 @@ public boolean canConnect(String ip) {
...
+            server.send(allMoves);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
 
+        allMoves = "";
+        String moves = server.getEnemyShips();
+        String[] movesArr = moves.split(",;");
+        for (int i = 0; i < movesArr.length; i++) {
+            String[] coordinates = movesArr[i].split(",");
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java]: maybe move to method

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java [hunk 11]
```diff
@@ -37,15 +37,16 @@
                 typesOfShips.poll();
                 if (typesOfShips.isEmpty()) {
                     socketServer.passAllShips(playerBoard.getAllpositions());
-                    //todo: method to place ships is called here!!!!!!
-                    running = placeShipsRandomly();
+                    //running = placeShipsRandomly();
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java]: remove comments

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java [hunk 12]
```diff
@@ -61,24 +62,41 @@ private void shipOrientation(MouseEvent event) {
...
-            int y = random.nextInt(9);
-            Point2D point2D = Point2D.of(x, y);
-            Ship ship = new Ship(shipCreator.createBattleShip(Collections.singletonList(point2D)));
-            if (enemyBoard.isShipPositionValid(ship, new Cell(point2D))) {
-                numberOfShipTypes--;
+    boolean placeShipsRandomly(String shipsString) {
+        String[] ships = shipsString.split(",\\|");
+        for (String shipStr : ships) {
+            List<Point2D> point2DOfShip = new ArrayList<>();
+            //System.out.println(shipStr);
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java]: remove comments

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java [hunk 13]
```diff
@@ -61,24 +62,41 @@ private void shipOrientation(MouseEvent event) {
...
+                int x = Integer.valueOf(coords[i]);
+                int y = Integer.valueOf(coords[i + 1]);
+                point2DOfShip.add(Point2D.of(x, y));
             }
+            shipCreator = createShip(coords);
+            Ship ship = new Ship(shipCreator.createBattleShip(point2DOfShip));
+            enemyBoard.isShipPositionValid(ship, new Cell(point2DOfShip.get(0)));
+            //System.out.println(Arrays.toString(coords));
         }
+        //return enemyBoard.allShips;
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java]: remove

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java [hunk 14]
```diff
@@ -61,24 +62,41 @@ private void shipOrientation(MouseEvent event) {
...
     }
-    //todo: dominik
+
+    String createShip(String[] coords, boolean v) {
+        if (coords.length > 2 && Integer.valueOf(coords[0]).equals(Integer.valueOf(coords[2])))
+            return "vertical";
+//            return new ShipCreator(new VerticalShipFactory());
+        else
+        {
+//            return new ShipCreator(new HorizontalShipFactory());
```
Comment (rivalez) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/ShipPlacer.java]: comments

File: battleship-server/src/main/java/com.epam.solid.nie/server/server/ShipSocketServer.java [hunk 15]
```diff
@@ -13,25 +16,65 @@
...
+        currentPlayer = first;
+
+        System.out.println("Initialized");
+    }
+
+    @Override
+    public void play() throws IOException {
+        String move = currentPlayer.makeMove();
+        System.out.println(players.indexOf(currentPlayer) + ":" + move);
+        if(move.equals("Q"))
```
Comment (rivalez) [inline battleship-server/src/main/java/com.epam.solid.nie/server/server/ShipSocketServer.java]: move Q to const

File: battleship-server/src/main/java/com.epam.solid.nie/server/server/Player.java (hunk 7 above)
Comment (Ozymadias) [reply battleship-server/src/main/java/com.epam.solid.nie/server/server/Player.java]: any suggestion?

**Pull request merged**